privtrak.go

```diff
@@ -393,7 +409,7 @@ func (m *ptMiddleware) HandleAnnounce(ctx context.Context, req *bittorrent.Annou
 		shard.deltas = append(shard.deltas, *delta)
 	}
 
-	expiredDeltas := stats.removeExpired(m.cfg.PeerLifetime)
+	expiredDeltas := stats.removeExpired(m.cfg.PeerLifetime, id) //TODO: double check passing this ID
```

Comment:
why are we calling this function on every announce? shouldn't it be called anyway in the GC flush goroutine? is it just to increase the resolution of checking for expired peers?

Reply:
Well, I added it on every announce because it has the potential of cleaning up some garbage wile using little time. It makes it so that the GC routine takes less time, because a lot of the stuff is already cleaned up.
I don't have real-world metrics or anything, but I feel like this is not that bad. We already have the shard locked, so we might as well have a look at this.

Reply:
does including the announceRequest ID into that removeExpired function call make sense though? won't that be assigning the requests ID to a bunch of unrelated deltas that happen to be getting expired at the same time?

Reply:
adding the ID makes sense now, because we need it for the emitted deltas, and the function itself can't get it easily. It won't assign the ID to the wrong deltas, because `removeExpired` runs on just one `userStats`, namely the one we got by the ID from the request.